### PR TITLE
Fixing issues with advanced_indexing_kernel

### PR DESCRIPTION
**Pull request by ipdemes**
There are 2 issues fixed in this PR:
1) Fix CUDA kernel logic
2) Fix incorrect use of ThreadLocalStorage in the OpenMP kernel

### CODE REVIEW

File: src/cunumeric/index/advanced_indexing_omp.cc [hunk 1]
```diff
@@ -78,8 +50,15 @@ struct AdvancedIndexingImplBody<VariantKind::OMP, CODE, DIM, OUT_TYPE> {
         sizes[tid] += static_cast<int64_t>(index[p] && ((idx + 1) % skip_size == 0));
       }
     }  // end of parallel
-    size_t size = thrust::reduce(thrust::omp::par, sizes.begin(), sizes.end(), 0);
-    thrust::exclusive_scan(thrust::omp::par, sizes.begin(), sizes.end(), offsets.begin());
+    size_t size = 0;
+    for (auto idx = 0; idx < max_threads; ++idx) {
+      size += sizes[idx];
+      offsets[idx] = sizes[idx];
```
Comment (rohany) [inline src/cunumeric/index/advanced_indexing_omp.cc:56]: You're already computing a reduction over `sizes`, so you might as well just do the exclusive prefix sum by setting `offsets[idx] = size` and moving to before you increment size.

Comment (ipdemes) [reply src/cunumeric/index/advanced_indexing_omp.cc]: that is right. Thank you!

**Comment by rohany** [comment]
The PR looks good to me, but @manopapad will give the final review.

File: src/cunumeric/index/advanced_indexing_omp.cc [hunk 2]
```diff
@@ -98,7 +74,8 @@ struct AdvancedIndexingImplBody<VariantKind::OMP, CODE, DIM, OUT_TYPE> {
 
     const auto max_threads = omp_get_max_threads();
     const size_t volume    = rect.volume();
-    ThreadLocalStorage<int64_t> offsets(max_threads);
+    auto kind    = CuNumeric::has_numamem ? Memory::Kind::SOCKET_MEM : Memory::Kind::SYSTEM_MEM;
+    auto offsets = create_buffer<int64_t>(max_threads, kind);
```
Comment (manopapad) [inline src/cunumeric/index/advanced_indexing_omp.cc:78]: Was there an issue with `offsets` being a `ThreadLocalStorage` object, after you removed the thrust calls?

Comment (ipdemes) [reply src/cunumeric/index/advanced_indexing_omp.cc]: Nothing is wrong with it since I don't use Thrust anymore. Replacing buffer with ThreadLocalStorage

**Review by manopapad — APPROVED** [review]

**Pull request merged**